Cleanup installs happening in the github workflows

bsweger opened the pull request:
Addresses #404 

The scope of these commits is making sure that all required dependencies are in the package's DESCRIPTION instead of being explicitly installed as a workflow step (for example, DT is required to build the vignettes).

Additionally, there's some cleanup to remove workflow-based installs if they aren't being used and/or reducing the footprint (e.g., explicitly use `ggplot2` in the vignette rather than pull in the entire tidyverse).

Longer term, we have a goal to make these workflows better align to the newer ones used in the hubverse (#406 )

**note:** there are several smaller commits so it's easier to revert any breaking changes...happy to squash after approval to tidy up the commit history

nickreich commented:
@bsweger I'm assuming you will ping me when this is ready for a full review, and am holding off on a review until then.

bsweger commented on .github/workflows/pkgdown.yaml:
```diff
@@ -21,11 +21,11 @@ jobs:
-          devtools::install_github("epiforecasts/scoringutils")
-          remotes::install_deps(dependencies = NA)
+          install.packages(c("devtools","pkgdown"))
+          devtools::install_github("eepiforecasts/scoringutils@v1.2.2")
```
I assumed the `scoringutils` version here should match the changes made in https://github.com/reichlab/covidHubUtils/pull/400

bsweger commented on vignettes/covidHubUtils-overview.Rmd:
```diff
@@ -67,12 +67,12 @@ We have identified two central use cases for downloading data using `load_foreca
 library(covidHubUtils)
 library(doParallel)
-library(tidyverse)
+library(ggplot2)
```
Another place where we can be more specific about the required library to avoid a big tidyverse install.

bsweger commented on vignettes/covidHubUtils-overview.Rmd:
````diff
@@ -433,7 +433,6 @@ The inputs to the `scored_forecasts()` include a `forecasts` data frame created
 
 The `scoringutils` package provides a collection of metrics and proper scoring rules that make it simple to score forecasts against the true observed values. 
 ``` {r}
-# devtools::install_github("epiforecasts/scoringutils@v1.2.2")
````
`scoringutils` is already installed as part of `covidHubUtils`

bsweger commented:
@nickreich ready for review!

elray1 commented on .github/workflows/pkgdown.yaml:
```diff
@@ -21,11 +21,11 @@ jobs:
-          devtools::install_github("epiforecasts/scoringutils")
-          remotes::install_deps(dependencies = NA)
+          install.packages(c("devtools","pkgdown"))
+          devtools::install_github("eepiforecasts/scoringutils@v1.2.2")
```
```suggestion
          devtools::install_github("epiforecasts/scoringutils@v1.2.2")
```

bsweger replied:
Great catch, thank you! 😅

elray1 commented on DESCRIPTION:
```diff
@@ -80,18 +81,20 @@ Imports:
     parallel,
     DBI,
-    RSQLite
+    RSQLite,
+    tidyr
```
It looks like tidyr was already listed above, on new line 58?
```suggestion
    RSQLite
```

bsweger replied:
ah--good call!

await my forthcoming linting and styling PR in which we shall alphabetize the list of imports

elray1 reviewed (changes requested):
Just a couple of minor changes in comments.

A bigger point is that although the github action definition updates looked plausible to me, I don't know of a way to test that they work without actually merging the changes in and giving them a run... do you?

bsweger commented:
> A bigger point is that although the github action definition updates looked plausible to me, I don't know of a way to test that they work without actually merging the changes in and giving them a run... do you?

@elray1 `r_cmd_check.yaml` and `pr_unittest.yaml` run when a PR opens, so I made sure they worked before opening the PR for review (however, the tests started failing after I committed your suggested changes, so I'll look at that).

is `pkgdown.yaml` something we can run on a PR and then again on merge to main?

I recall that github actions have some capacity for ad-hoc runs, but I haven't looked into it yet.

nickreich reviewed (approved):
It feels like I'm breaking some GitHub code of conduct to give an approving review to a PR that is not passing a check, but I did get a few minutes to look at this and it the changes look good to me. So once the checks are passing I don't have other concerns.

bsweger commented:
@elray1 I believe I found and fixed the issue that was causing the tests to fail (that said, I'm not sure why they were passing before merging in your requested changes (which seem unrelated..and I reverted them to just see if the tests would start passing again, and they did! :tableflip:)

Anyway, everything is green now, and I squashed the commits so the history is a bit cleaner. Can you re-review when you have a chance? Thanks!
